internal/configs/configschema/path.go

```diff
@@ -1,19 +1,27 @@
 package configschema
 
 import (
+	"fmt"
+
 	"github.com/zclconf/go-cty/cty"
 )
 
 // AttributeByPath looks up the Attribute schema which corresponds to the given
 // cty.Path. A nil value is returned if the given path does not correspond to a
 // specific attribute.
-// TODO: this will need to be updated for nested attributes
 func (b *Block) AttributeByPath(path cty.Path) *Attribute {
 	block := b
-	for _, step := range path {
+	for i, step := range path {
 		switch step := step.(type) {
 		case cty.GetAttrStep:
 			if attr := block.Attributes[step.Name]; attr != nil {
+				// If the Attribute is defined with a NestedType and there's
+				// more to the path, descend into the NestedType
+				if attr.NestedType != nil && i < len(path)-1 {
+					return attr.NestedType.AttributeByPath(path)
```

Comment:
The path is not being trimmed here, passing the already traversed elements to `Object.AttributeByPath`. If the nested type contains any attributes matching the prior steps, it will continue down the wrong path

Reply:
Ah, good point, I hadn't thought that through when I left the trimming off.